### ISSUE TITLE
Training with pretrained-backbone

### ISSUE THREAD

**Issue by littlejiumi** [Arthur151/ROMP#125]
Thank you very much for sharing the code. 
I used the pretrained backbone (trained_models/pretrain_hrnet.py) you provided, but found that the result is not ideal, worse than train from scratch, here are my logs:
[ROMP-from-scratch.log](https://github.com/Arthur151/ROMP/files/7767004/ROMP-from-scratch.log)
[ROMP-pretrained.log](https://github.com/Arthur151/ROMP/files/7767006/ROMP-pretrained.log)
By the way, the pretrained backbone you provide may need to change the key name (plus 'backbone.') in order to run normally. current_model_statedict.keys(): 
```
backbone.conv1.weight
backbone.bn1.weight
backbone.bn1.bias.... 
```
and pretrained_model_statedict.keys():
```
conv1.weight
bn1.weight
bn1.bias
```
So I change base.py in L49 to 
`model = load_model(self.model_path, model, prefix='', drop_prefix='backbone.', fix_loaded=False)`

Looking forward to your reply！

**Comment by Arthur151** [comment]
Thanks for your report!
This is pretty weird. Could you please provide the complete log especially the initialization part before training started.

**Comment by littlejiumi** [comment]
TTThank you for your prompt reply!
[hrnet-pretrained.txt](https://github.com/Arthur151/ROMP/files/7767752/hrnet-pretrained.txt)
[hrnet.txt](https://github.com/Arthur151/ROMP/files/7767753/hrnet.txt)
here are the initialization part of logs

**Comment by Arthur151** [comment]
From the configuration, I think the problem is about the batch size.  
The batch size of hrnet.txt is 8. While the batch size of hrnet-pretrained.txt is 32. I use 64 for training. 
Please refer to [this issue](https://github.com/Arthur151/ROMP/issues/121). This user trained ROMP with batch size 48 and achieve better results. 
Maybe our discussion can help you too.

**Comment by littlejiumi** [comment]
> Thank you very much for sharing the code. I used the pretrained backbone (trained_models/pretrain_hrnet.py) you provided, but found that the result is not ideal, worse than train from scratch, here are my logs: [ROMP-from-scratch.log](https://github.com/Arthur151/ROMP/files/7767004/ROMP-from-scratch.log) [ROMP-pretrained.log](https://github.com/Arthur151/ROMP/files/7767006/ROMP-pretrained.log) By the way, the pretrained backbone you provide may need to change the key name (plus 'backbone.') in order to run normally. current_model_statedict.keys():
> 
> ```
> backbone.conv1.weight
> backbone.bn1.weight
> backbone.bn1.bias.... 
> ```
> 
> and pretrained_model_statedict.keys():
> 
> ```
> conv1.weight
> bn1.weight
> bn1.bias
> ```
> 
> So I change base.py in L49 to `model = load_model(self.model_path, model, prefix='', drop_prefix='backbone.', fix_loaded=False)`
> 
> Looking forward to your reply！

 I try batchsize==32 both in "train from scratch" and "train on Pretrained hrnet"(I did not save the initialization part log, sorry). The result of the former one is reasonable and close to the results you gave, but  the loss of the latter is higher than the former, which is weird. I want to know whether
1)"Train all layers, except: ['_result_parser.params_map_parser.smpl_model.betas']" is right in "train on Pretrained hrnet"
2) the learning rate and other training tricks on "train on Pretrained hrnet"

OR 

can you give me your log file for "train on Pretrained model"?
Thank you so much!

**Comment by Arthur151** [comment]
About your question, 
1) Yes, "Train all layers, except: ['_result_parser.params_map_parser.smpl_model.betas']" is right.
2) the learning rate is always 5e-5 for batch size 64 in my experiments. I think one trick that may help you. As I said above, I think the problem is batch size. You may increase the batch size via freezing the backbone and train the head first. And then train the whole model togather. About  freezing the weight of backbone, [this function](https://github.com/Arthur151/ROMP/blob/ed01ce3e9a286436f5251ed9a4aaddbc768f0e07/romp/lib/utils/train_utils.py#L105) might be helpful. You can also un-freeze the higher layers of backbone only to keep the batch size staill being 64 in second training stage.

**Comment by ZhengdiYu** [comment]
> he batch size of hrne

@littlejiumi Hi, can you confirm that you didn't use '**`pretrain_hrnet.pkl`**' when **training-from-scratch** ? I would like to know that **how long** did you train the model **from scratch** to get this results?

Recently, the author and I have both tried to comment 'self.backbone.load_pretrain_params()' in modelv1.py to train from scratch. However both of us can only get a very bad results compare to training from pretrain_hrnet.pkl in the first several epochs.  For example, if we train from pre-trained backbone, we can get **MPJPE** around 91 within first two epochs, but if we train from scratch, it's still larger than 150 after several epochs.
